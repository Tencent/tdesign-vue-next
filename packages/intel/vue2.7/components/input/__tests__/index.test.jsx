/* eslint-disable no-unused-vars */
/* eslint-disable @typescript-eslint/no-unused-vars */
import Vue from 'vue';
import { mount } from '@vue/test-utils';
import { Input } from '@/src/input/index.ts';

describe('Input', () => {
  describe(':props', () => {
    it(':value', () => {
      const wrapper = mount({
        render() {
          return <Input value={'text'} />;
        },
      });
      const inputElemWrapper = wrapper.find('input');
      expect(inputElemWrapper.element.value).toEqual('text');
    });

    it(':value(controlled)', async () => {
      const wrapper = mount({
        render() {
          return <Input value={'text'} />;
        },
      });
      const inputElemWrapper = wrapper.find('input');
      inputElemWrapper.setValue('text1');
      await Vue.nextTick();
      expect(inputElemWrapper.element.value).toEqual('text');
    });

    it(':default-value', async () => {
      const wrapper = mount({
        render() {
          return <Input default-value={'text'} />;
        },
      });
      const inputElemWrapper = wrapper.find('input');
      inputElemWrapper.setValue('text1');
      await Vue.nextTick();
      expect(inputElemWrapper.element.value).toEqual('text1');
    });

    it(':disabled', () => {
      const wrapper = mount({
        render() {
          return <Input disabled={true} />;
        },
      });
      expect(wrapper.element).toMatchSnapshot();
    });

    it(':readonly', () => {
      const wrapper = mount({
        render() {
          return <Input readonly={true} />;
        },
      });
      expect(wrapper.element).toMatchSnapshot();
    });

    it(':autocomplete', () => {
      const wrapper = mount({
        render() {
          return <Input autocomplete={true} />;
        },
      });
      expect(wrapper.element).toMatchSnapshot();
    });

    it(':prefix-icon', () => {
      const wrapper = mount({
        render() {
          return <Input prefix-icon={(h) => <i class="icon"></i>} />;
        },
      });
      expect(wrapper.element).toMatchSnapshot();
    });

    it(':suffix-icon', () => {
      const wrapper = mount({
        render() {
          return <Input suffix-icon={(h) => <i class="icon"></i>} />;
        },
      });
      expect(wrapper.element).toMatchSnapshot();
    });

    it(':size', () => {
      const wrapper = mount({
        render() {
          return <Input size={'large'} />;
        },
      });
      expect(wrapper.element).toMatchSnapshot();
    });
  });

  describe('$attrs', () => {
    it('input attrs should pass to input element', () => {
      const wrapper = mount({
        render() {
          return <Input name="password" placeholder="please text" maxLength="20" type="password" />;
        },
      });
      expect(wrapper.element).toMatchSnapshot();
    });
  });

  describe('@event', () => {
    it('@change', async () => {
      const fn = vi.fn();
      const wrapper = mount({
        render() {
          return <Input on={{ change: fn }} change={fn} />;
        },
      });
      const inputWrapper = wrapper.findComponent(Input);
      const inputElemWrapper = wrapper.find('input');
      inputElemWrapper.setValue('text');
      await inputWrapper.vm.$nextTick();
      // TODO emitted return undefined in vue 2.7
      // expect(inputWrapper.emitted().change).toBeTruthy();
      expect(fn).toBeCalled();
    });

    it('@blur', () => {
      const fn = vi.fn();
      const wrapper = mount({
        render() {
          return <Input onBlur={fn} />;
        },
      });
      const inputWrapper = wrapper.findComponent(Input);
      const inputElemWrapper = wrapper.find('input');
      inputElemWrapper.trigger('blur');
      // TODO emitted return undefined in vue 2.7
      // expect(inputWrapper.emitted().blur).toBeTruthy();
      expect(fn).toBeCalled();
    });

    it(':onKeydown', async () => {
      const fn = vi.fn();
      const wrapper = mount({
        render() {
          return <Input onKeydown={fn} />;
        },
      });
      const input = wrapper.find('.t-input input');
      await input.trigger('keydown');
      expect(fn).toBeCalled();
    });
  });

  describe('methods', () => {
    it('blur', () => {
      const fn = vi.fn();
      const wrapper = mount({
        render() {
          return <Input onBlur={fn} />;
        },
      });
      const inputWrapper = wrapper.findComponent(Input);
      const inputElemWrapper = wrapper.find('input');
      inputWrapper.vm.focus();
      inputWrapper.vm.blur();
      inputElemWrapper.trigger('blur');

      // TODO emitted return undefined in vue 2.7
      // expect(inputWrapper.emitted().blur).toBeTruthy();
      expect(fn).toBeCalled();
    });
  });
});
