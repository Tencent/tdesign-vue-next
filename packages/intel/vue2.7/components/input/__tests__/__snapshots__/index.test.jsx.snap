// Vitest Snapshot v1

exports[`Input > $attrs > input attrs should pass to input element 1`] = `
<div
  class="t-input__wrap"
>
  <div
    class="t-input t-input--suffix"
    maxlength="20"
  >
    <input
      autocomplete=""
      class="t-input__inner"
      name="password"
      placeholder="please text"
      type="password"
      unselectable="off"
    />
    <span
      class="t-input__suffix t-input__suffix-icon"
    >
      <svg
        class="t-icon t-icon-browse-off t-input__suffix-clear"
        fill="none"
        height="1em"
        viewBox="0 0 26 24"
        width="1em"
      >
        <path
          d="M4 1.59l6.17 6.17 7.07 7.07L23.41 21 22 22.41l-2.97-2.96A12.5 12.5 0 011.08 12.3L1 12l.1-.3c.77-2.4 2.24-4.5 4.18-6.02L2.59 3 4 1.59zM6.7 7.1A10.53 10.53 0 003.1 12a10.5 10.5 0 0014.45 5.97l-1.8-1.8a5 5 0 01-6.93-6.93L6.7 7.11zm3.6 3.6a3 3 0 004 4l-4-4zM13 5c-.58 0-1.14.05-1.7.14l-.98.16L10 3.32l.99-.16A12.5 12.5 0 0124.9 11.7l.1.31-.1.3c-.41 1.3-1.03 2.5-1.82 3.58l-.59.8-1.61-1.18.59-.8c.6-.82 1.08-1.73 1.42-2.7A10.5 10.5 0 0013 5zm.51 1.93l.96.29a5 5 0 013.31 3.31l.3.96-1.92.58-.3-.95a3 3 0 00-1.98-1.99l-.95-.3.58-1.9z"
          fill="currentColor"
        />
      </svg>
    </span>
  </div>
</div>
`;

exports[`Input > :props > :autocomplete 1`] = `
<div
  class="t-input__wrap"
>
  <div
    class="t-input"
  >
    <input
      autocomplete="true"
      class="t-input__inner"
      placeholder="请输入"
      type="text"
      unselectable="off"
    />
  </div>
</div>
`;

exports[`Input > :props > :disabled 1`] = `
<div
  class="t-input__wrap"
>
  <div
    class="t-input t-is-disabled"
  >
    <input
      autocomplete=""
      class="t-input__inner"
      disabled="disabled"
      placeholder="请输入"
      type="text"
      unselectable="off"
    />
  </div>
</div>
`;

exports[`Input > :props > :prefix-icon 1`] = `
<div
  class="t-input__wrap"
>
  <div
    class="t-input t-input--prefix"
  >
    <span
      class="t-input__prefix t-input__prefix-icon"
    >
      <i
        class="icon"
      />
    </span>
    <input
      autocomplete=""
      class="t-input__inner"
      placeholder="请输入"
      type="text"
      unselectable="off"
    />
  </div>
</div>
`;

exports[`Input > :props > :readonly 1`] = `
<div
  class="t-input__wrap"
>
  <div
    class="t-input t-is-readonly"
  >
    <input
      autocomplete=""
      class="t-input__inner"
      placeholder="请输入"
      readonly="readonly"
      type="text"
      unselectable="on"
    />
  </div>
</div>
`;

exports[`Input > :props > :size 1`] = `
<div
  class="t-input__wrap"
>
  <div
    class="t-input t-size-l"
  >
    <input
      autocomplete=""
      class="t-input__inner"
      placeholder="请输入"
      type="text"
      unselectable="off"
    />
  </div>
</div>
`;

exports[`Input > :props > :suffix-icon 1`] = `
<div
  class="t-input__wrap"
>
  <div
    class="t-input t-input--suffix"
  >
    <input
      autocomplete=""
      class="t-input__inner"
      placeholder="请输入"
      type="text"
      unselectable="off"
    />
    <span
      class="t-input__suffix t-input__suffix-icon"
    >
      <i
        class="icon"
      />
    </span>
  </div>
</div>
`;
